components/style/values/specified/position.rs

```diff
@@ -32,27 +32,25 @@ pub struct Position {
 
 impl ToCss for Position {
     fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
-        let mut space_present = false;
-        if let Some(horiz_key) = self.horizontal.keyword {
-            try!(horiz_key.to_css(dest));
-            try!(dest.write_str(" "));
-            space_present = true;
-        };
-        if let Some(ref horiz_pos) = self.horizontal.position {
-            try!(horiz_pos.to_css(dest));
-            try!(dest.write_str(" "));
-            space_present = true;
-        };
-        if let Some(vert_key) = self.vertical.keyword {
-            try!(vert_key.to_css(dest));
-            space_present = false;
-        };
-        if let Some(ref vert_pos) = self.vertical.position {
-            if space_present == false {
-                try!(dest.write_str(" "));
-            }
-            try!(vert_pos.to_css(dest));
-        };
+        let mut is_keyword_needed = false;
+        let position_x = &self.horizontal;
+        let position_y = &self.vertical;
+
+        if (position_x.keyword.is_some() && position_x.position.is_some()) ||
+           (position_y.keyword.is_some() && position_y.position.is_some()) {
+            is_keyword_needed = true;
+        }
+
+        if is_keyword_needed {
+            position_x.to_css_with_keyword(dest)?;
```

Comment:
I think you can just inline `to_css_with_keyword` into this function (as I mentioned before), since that is not used anywhere else.